prac_05/wimbledon.py

```diff
@@ -0,0 +1,55 @@
+"""
+CP1404Practical
+Wimbledon Program
+"""
+
+FILE_NAME = "wimbledon.csv"
+COUNTRY_INDEX = 1
+CHAMPION_INDEX = 2
+
+
+def main():
+    """Read csv file and display details about Wimbledon champions and countries."""
+    records = get_records(FILE_NAME)
+    champion_to_win, countries = process_records(records)
+    display_results(champion_to_win, countries)
+
+
+def display_results(champion_to_win, countries):
+    """Display champions and countries"""
+    print("Wimbledon Champions: ")
+    for champion, number_of_wins in champion_to_win.items():
+        print(champion, number_of_wins)
+    print(f"\nThese {len(countries)} countries have won Wimbledon:")
+    print(", ".join(sorted(countries)))
+
+
+def process_records(records):
```

Comment:
This function could be breaking the SRP, as it is doing a lot of work.

Reply:
I see what you mean, but I believe that the solution does the same thing...

Reply:
I see, hahaha. I'm not one to argue with the solution, but I feel that if this program ever got expanded to show more data (e.g. as seen in the extension task) this function would turn very very big, which, I am not sure if that is a problem or not... and I guess if it did start getting really big you could turn each of the processing tasks into a function within the function.

Reply:
Processing once... do all the things while looping over the data should be "one" job and more efficient than looping over the data once for each kind of processing, then doing the same thing again for a different kind of processing.